src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.cs

```diff
@@ -52,13 +51,15 @@ internal static ImmutableArray<TypeSymbolWithAnnotations> GetAnonymousTypeProper
             Debug.Assert(type.IsAnonymousType);
             var anonymous = (AnonymousTypePublicSymbol)type;
             var fields = anonymous.TypeDescriptor.Fields;
-            var types = ArrayBuilder<TypeSymbolWithAnnotations>.GetInstance(fields.Length);
-            for (int i = 0; i < fields.Length; i++)
-            {
-                types.Add(fields[i].Type);
-            }
+            return fields.SelectAsArray(f => f.Type);
+        }
 
-            return types.ToImmutableAndFree();
+        internal static PropertySymbol AsMember(PropertySymbol property, NamedTypeSymbol newOwner)
```

Comment:
>AsMember [](start = 39, length = 8)

Given that all other `AsMember` methods I looked at have strict assertions, I think we should name this method differently.

The assertions in other `AsMember` methods:
```C#
            Debug.Assert(this.IsDefinition);
            Debug.Assert(ReferenceEquals(newOwner.OriginalDefinition, this.ContainingSymbol.OriginalDefinition));
```

Reply:
Removed method.

---
In reply to: [259057071](https://github.com/dotnet/roslyn/pull/33562#discussion_r259057071) [](ancestors = 259057071)